JsonLd/Resource.php

```diff
@@ -272,6 +312,90 @@ public function getRouteCollection()
     }
 
     /**
+     * {@inheritdoc}
+     */
+    public function getCollectionRoute()
+    {
+        if (!$this->collectionRoute) {
+            $this->getRouteCollection();
```

Comment:
This is not the same thing. One is a string containing the route, the other contains an instance of Symfony router `RouteCollection`. A better name is welcome :)